File: server/src/modules/rental-requests/domain/entities/contract.entity.ts
export type ContractStatus =
  | 'draft'
  | 'sent'
  | 'signed'
  | 'cancelled'
  | 'finished';

export class ContractEntity {
  constructor(
    public readonly id: string,
    public readonly rentalRequestId: string | null,
    public readonly roomId: string,
    public readonly tenantId: string,
    public readonly landlordId: string,
    public readonly startDate: string,
    public readonly endDate: string,
    public readonly monthlyRent: string,
    public readonly deposit: string,
    public readonly status: ContractStatus,
    public readonly terms: string | null,
    public readonly sentAt: Date | null,
    public readonly signedAt: Date | null,
    public readonly cancelledAt: Date | null,
    public readonly finishedAt: Date | null,
    public readonly createdAt: Date,
    public readonly updatedAt: Date,
  ) {}
}

File: client/packages/domain/lib/src/rental_request/entities/contract_entity.dart
enum ContractStatus { draft, sent, signed, cancelled, finished }

abstract class ContractEntity {
  ContractEntity({
    required this.id,
    required this.roomId,
    required this.tenantId,
    required this.landlordId,
    required this.startDate,
    required this.endDate,
    required this.monthlyRent,
    required this.deposit,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.rentalRequestId,
    this.terms,
    this.momoNumber,
    this.sentAt,
    this.signedAt,
    this.cancelledAt,
    this.finishedAt,
  });

  final String id;
  final String? rentalRequestId;
  final String roomId;
  final String tenantId;
  final String landlordId;
  final String startDate;
  final String endDate;
  final double monthlyRent;
  final double deposit;
  final ContractStatus status;
  final String? terms;
  final String? momoNumber;
  final String? sentAt;
  final String? signedAt;
  final String? cancelledAt;
  final String? finishedAt;
  final String createdAt;
  final String updatedAt;
}
